modules/apigee_edge_apidocs/src/Form/ApiDocUpdateSpecForm.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\apigee_edge_apidocs\Form;
+
+use Drupal\Core\Entity\ContentEntityConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Messenger\MessengerTrait;
+use Drupal\Core\Url;
+
+/**
+ * Class ApiDocUpdateSpecForm.
+ */
+class ApiDocUpdateSpecForm extends ContentEntityConfirmFormBase {
+
+  use MessengerTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $operation = 'update_spec';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Are you sure you want to update the OpenAPI specification
+     file from URL on API Doc %name?', [
+       '%name' => $this->entity->label(),
+      ]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    return new Url('entity.apidoc.collection');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDescription() {
+    return $this->t('This will replace the current OpenAPI specification file.
+     This action cannot be undone.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    /* @var \Drupal\apigee_edge_apidocs\Entity\ApiDocInterface $entity */
+    $entity = $this->getEntity();
+    $status = $entity->updateOpenApiSpecFile(TRUE, TRUE);
```

Comment:
If the Open API spec fetcher becomes a service, the design here also changes. For example, the new service could provide two public methods:
a) for fetching Open API specification (and saving it to a new API doc entity in the API doc entity storage)
b) for updating Open API specification on a given API document entity *if* it has been created (its latest revision has been created) from an URL import and *if* the Open API file has changed on the remote server.

Although, having an Open API fetcher and an API documentation updater service would be a more robust design. The fetcher would only handle the fetching and the updater would update speciation on all or a give API documentation entity.

Reply:
The Open API spec fetcher becomes is now a service.